components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth/OAuthUtil.java

```diff
@@ -1257,10 +1258,21 @@ public static void removeAuthzGrantCacheForUser(String userName, UserStoreManage
              Only the tokens and auth codes issued for openid scope should be removed from the cache, since no
              claims are usually cached against tokens or auth codes, otherwise.
              */
+
+            // Retrieve the tokens and auth codes associated with usernames without domain.
             accessTokenDOSet = OAuthTokenPersistenceFactory.getInstance().getAccessTokenDAO()
                     .getAccessTokensByUserForOpenidScope(authenticatedUser);
             authorizationCodeDOSet = OAuthTokenPersistenceFactory.getInstance()
                     .getAuthorizationCodeDAO().getAuthorizationCodesByUserForOpenidScope(authenticatedUser);
+            
+            // Retrieve the tokens and auth codes associated with domain-qualified usernames.
+            if (!userName.contains(UserCoreConstants.DOMAIN_SEPARATOR)) {
```

Comment:
For a normal case where username is not domain qualified, same operations gonna run twice now.  IS that fine

CC: @NipuniBhagya

Reply:
isn't there a possibility of checking multi attribute login enabled at this point?

Reply:
`For a normal case where username is not domain qualified, same operations gonna run twice now. IS that fine` - That's one of the issue with this approach.

Reply:
> isn't there a possibility of checking multi attribute login enabled at this point?

In that case we have to guarantee that there are no other flows that will have the same behaviour as SMS OTP with multi attribute login.

Reply:
Kindly refer to the email : `[Asgardeo] Refresh Token Grant Returns JWT Access Tokens with Stale Claims`, where I have pointed out in detailed why it is better to check for both formats simultaneously.